java/src/main/java/org/wildfly/openssl/OpenSSLContextSPI.java

```diff
@@ -175,17 +191,31 @@ private synchronized void init(KeyManager[] kms, TrustManager[] tms) throws KeyM
             return;
         }
 
+        // a single subject can have multiple certificates for different algorithms, as
+        // aliases are required to be unique, the subject is the next best thing to establish
+        // some form of grouping, as a single context can have multiple certificates
+        // for different algorithms
+        final Map<String, Long> subjectToSSLContextMap = new LinkedHashMap<>();
+
+        // this simple map is used later on during certificate selection in the SNICallback,
+        // as a single ssl ctx can have multiple certificate, and SNI uses a requested
+        // hostname to allow the server to choose the certificate, we flatten everything
+        final Map<ArrayList<String>, Long> x509CertificateToSSLContextMap = new LinkedHashMap<>();
```

Comment:
Just wondering if instead of this approach, we could make it possible to directly configure a `Map<SNIMatcher, SSLContext>` instead.

Reply:
refactored the same, hope you like it better now :)